src/utils/server/basescan/getBaseETHBalances.ts

```
@@ -0,0 +1,40 @@
+import * as Sentry from '@sentry/nextjs'
+
+import { fetchReq } from '@/utils/shared/fetchReq'
+import { requiredEnv } from '@/utils/shared/requiredEnv'
+
+const BASESCAN_API_KEY = requiredEnv(process.env.BASESCAN_API_KEY, 'process.env.BASESCAN_API_KEY')
+
+type GetBaseETHBalancesResponse = {
+  status: string
+  message: string
+  result: Array<{
+    account: string
+    balance: string
+  }>
+}
+
+/**
+ * Returns the wei balance of the given Ethereum addresses.
+ *
+ * @param cryptoAddresses
+ * @returns
+ */
+export async function getBaseETHBalances(cryptoAddresses: string[]) {
+  const url = `https://api.basescan.org/api?module=account&action=balancemulti&tag=latest&address=${cryptoAddresses.toString()}&apikey=${BASESCAN_API_KEY}`
+  try {
+    const httpResp = await fetchReq(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+    return (await httpResp.json()) as GetBaseETHBalancesResponse
+  } catch (error) {
+    Sentry.captureException(error, {
+      tags: { domain: 'getBaseETHBalances' },
+      extra: { url },
+    })
+    throw error
```

Do we need to throw an error here? If so, can we wrap the call to this in a try/catch or do we expect the flow to be unrecoverable?

---

@kl-cb I am throwing an error here so that the Inngest function knows to re-attempt a specific step.

But now that I am thinking about the cron job a little more... given the pretty frequent cadence of the cron job, I think it's better if I just not try to retry any specific steps and just throw a nonretriable error for the whole Inngest function.

---

Gotcha, didn't know it would automatically do a retry if it errors out!